libs/libarchfpga/src/arch_util.cpp

```diff
@@ -549,8 +547,8 @@ void alloc_and_load_default_child_for_pb_type(t_pb_type* pb_type,
         copy->annotations[i].format = pb_type->annotations[i].format;
         copy->annotations[i].type = pb_type->annotations[i].type;
         copy->annotations[i].num_value_prop_pairs = pb_type->annotations[i].num_value_prop_pairs;
-        copy->annotations[i].prop = (int*)vtr::malloc(sizeof(int) * pb_type->annotations[i].num_value_prop_pairs);
-        copy->annotations[i].value = (char**)vtr::malloc(sizeof(char*) * pb_type->annotations[i].num_value_prop_pairs);
+        copy->annotations[i].prop = new int[pb_type->annotations[i].num_value_prop_pairs];
+        copy->annotations[i].value = new char*[pb_type->annotations[i].num_value_prop_pairs];
```

Comment:
If you follow @soheilshahrouz 's suggestion on a vector of pairs of ints/strings above this copy will become easier -- should turn into something like copy->annotations = pb_type->annotations since the vector/pair/string classes know how to properly do a deep-copy assignment.